lib/xiaomi.js

```diff
@@ -142,10 +143,10 @@ const buffer2DataObject = (meta, model, buffer) => {
     return dataObject;
 };
 
-const numericAttributes2Payload = (msg, meta, model, options, dataObject) => {
+const numericAttributes2Payload = async (msg, meta, model, options, dataObject) => {
```

Comment:
I would prefer to not make this method async, instead of `await callback();` can we use `.then`/`.catch`?

Reply:
Sorry this is a bit lengthy. 

Unfortunately, in this case I do need the whole function to wait for the callback to finish. Otherwise, if we wait on a local level (`case '247'` in this case): 
```js
case '247':
    {
        const dataObject247 = buffer2DataObject(meta, model, value);
        if (['CTP-R01'].includes(model.model)) {
            // execute pending soft switch of operation_mode, if exists
            const opModeSwitchTask = globalStore.getValue(meta.device, 'opModeSwitchTask');
            if (opModeSwitchTask) {
                const {callback, newMode} = opModeSwitchTask;
                callback()
                    .then((_) => {
                        payload.operation_mode = newMode;
                        globalStore.putValue(meta.device, 'opModeSwitchTask', null);
                    });
            } else {
                payload.operation_mode = {0: 'action_mode', 1: 'scene_mode'}[dataObject247[155]];
            }
        }
        const payload247 = numericAttributes2Payload(msg, meta, model, options, dataObject247);
        payload = {...payload, ...payload247};
    }
    break;
```
the result payload wouldn't be the same. The execution will continue without waiting, thus `operation_mode` isn't updated. 

`dataObject247` report is triggered by user "throw", only then the cube will respond to write command (the callback's doing), and we need to wait for the command to be delivered to update `operation_mode` to desired `newMode`. 
If we don't wait for it, the payload would not contain updated `operation_mode`,  and user interaction experience breaks. 

See this [demo](https://github.com/JJPro/CTP-R01-converter/wiki/Demos-%26-My-Flows#demo) for user interaction. 
See [more on throw](https://github.com/JJPro/CTP-R01-converter/wiki/throw) for the concept of how the behavior is implemented.